src/preamble.js

```diff
@@ -1669,6 +1669,46 @@ function lookupSymbol(ptr) { // for a pointer, print out all symbols that resolv
 var memoryInitializer = null;
 
 #if USE_PTHREADS
+
+// JS library / EM_ASM proxying
+
+function proxyToMainThread() {
+  var args = Array.prototype.slice.call(arguments);
```

Comment:
Can we avoid duplicating the `arguments` object here? That leads to temp JS garbage, but looks like we could iterate over `arguments.length` and `arguments[i]` directly?